utils-intervalrdd/src/main/scala/org/bdgenomics/utils/rangearray/Interval.scala

```diff
@@ -16,14 +16,14 @@
  * limitations under the License.
  */
 
-package org.bdgenomics.utils.intervaltree
+package org.bdgenomics.utils.rangearray
 
 /**
  * An interval is a 2-dimensional 0-based coordinate consisting of a closed start value and open end value.
  * Each 2-dimensional coordinate has a defined width.
  * This can be used to express a region of a genome, a transcript, a gene, etc.
  */
-trait Interval {
+trait Interval[T <: Interval[T]] extends Comparable[T] {
```

Comment:
Actually yeah, I just checked looked at some of the code below and I'm pretty sure that's why. If so, savvy! I like it, let's just document it in the scaladoc for the trait for posterity.